bundle/edu.gemini.itc.shared/src/main/java/edu/gemini/itc/shared/ConfigCreator.java

```diff
@@ -0,0 +1,361 @@
+            }
+        }
+        return result;
+    }
+
+    public ConfigCreatorResult createNifsConfig(NifsParameters nifsParams, int numExp) {
+        ConfigCreatorResult result = createCommonConfig(numExp);
+
+        for (Config step : result.getConfig()) {
+            step.putItem(ReadModeKey, (nifsParams.readMode()));
+            step.putItem(InstInstrumentKey, SPComponentType.INSTRUMENT_NIFS);
+
+            if (nifsParams.altair().isDefined()) {
+                AltairParameters altairParameters = nifsParams.altair().get();
+                step.putItem(AOGuideStarTypeKey, altairParameters.wfsMode());
+                step.putItem(AOSystemKey, SPComponentType.AO_ALTAIR);
+            }
+        }
+        return result;
+    }
+
+    public ConfigCreatorResult createGsaoiConfig(GsaoiParameters gsaoiParams, int numExp) {
+        ConfigCreatorResult result = createCommonConfig(numExp);
+        double sourceFraction = itcParams.observation().sourceFraction();
+        int stepNum = 0;
+        int numLargeOffsets = gsaoiParams.largeSkyOffset();
+        boolean error = false;
+
+        GuideOption g = StandardGuideOptions.Value.guide;
+        GuideOption p = StandardGuideOptions.Value.park;
+
+        List<GuideOption> guideStatusList = new ArrayList<>(numExp);
+
+        /**
+         * GSAOI offset overheads and sequence structure is different from all other instruments. In addition
+         * the unguided sky sequence for the case of sky offset is >5' is made into a separate observation in the OT,
+         * whereas in the ITC it should be included in the same observation sequence. So the implementation
+         * of the offset and guide status sequence is far from elegant...
+         *
+         * The overhead calculation is currently supported for source fraction 1 and 0.5. Number of on- and off-source steps must be equal.
+         *    1. on-source fraction = 1.0, large sky offsets = 0: Regular ABAB dithering, no sky offsets, all steps guided.
+         *    2. on-source fraction = 1.0, large sky offsets !=0: Warning, no overheads calculation (using large sky offsets implies some steps being taken off-source).
+         *    3. on-source fraction = 0.5, large sky offsets = 0: Sequence has obj-sky-sky-obj structure, with 4-point ABAB dithering at each position
+         *                   (the last block can have less steps depending on requested number of exposures). Sky is unguided. Offset to the sky and back = 120"
+         *    4. on-source fraction = 0.5, large sky offsets !=0: Sequence has obj-sky-obj-sky structure, with ABAB dithering at each position. Sky is unguided. Offset to the sky and back = 310". When changing from unguided to guided, the LGS reacquisition overhead is added.
+         *    5. on-source fraction !=1.0 && !=0.5: Warning, no overhead calculations.
+         *
+         */
+
+        if ((sourceFraction != 1.0) && (sourceFraction != 0.5)) {
```

Comment:
Comparing doubles for equality is kind of sketchy. I guess we have some confidence that this works.